gems/smithy-client/lib/smithy-client/shapes.rb

```diff
@@ -0,0 +1,213 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    # Represents shape types from Smithy Model
+    module Shapes
+      # A base shape that all shapes inherits from
+      class Shape
+        def initialize(options = {})
+          @shape_id = options[:shape_id]
+          @traits = options[:traits] || {}
+        end
+
+        # @return [String, nil]
+        attr_reader :shape_id
+
+        # @return [Hash<String, [Boolean, Hash, Integer, String]>]
```

Comment:
https://yardoc.org/types.html

I believe you need Hash{String => Boolean, Hash, Integer, String} if you want multi values.

Reply:
Whatever we choose, let's just be consistent. I think Hash<String, Object> is fine, since the value can be any number of things. I think RBS can be more specific about boolean/hash/integer/string.

Reply:
Agreed, going with `Hash<String, Object>`